src/components/Nav.js

```diff
@@ -1,28 +1,54 @@
-import React from 'react';
-
+import { useContext } from 'react';
 import { NavLink } from 'react-router-dom';
+import { MapCenterContext } from '../context/MapCenterContext';
+import FlyToMeButton from './FlyToMeButton';
+import ShareMyLocationButton from './ShareMyLocationButton';
+
+const Nav = () => {
+  const valueCenterMap = useContext(MapCenterContext);
+  const { trackingId, userLocationShared } = valueCenterMap;
+  const wasWatchingLocation = trackingId !== null;
 
-const Nav = () => (
-  <footer>
-    <nav>
-      <ul className="nav">
-        <li>
-          <NavLink exact to="/">
-            Map
-          </NavLink>
-        </li>
-        <li>
-          <NavLink to="/list">List</NavLink>
-        </li>
-        <li>
-          <NavLink to="/settings">Settings</NavLink>
-        </li>
-        <li>
-          <NavLink to="/help">Help</NavLink>
-        </li>
-      </ul>
-    </nav>
-  </footer>
-);
+  return (
+    <footer>
+      <nav>
+        <ul className="nav">
+          <li>
+            <NavLink exact to="/">
+              Map
+            </NavLink>
+          </li>
+          <li>
+            <NavLink to="/list">List</NavLink>
+          </li>
+          <li>
+            <NavLink to="/settings">Settings</NavLink>
+          </li>
+          <li>
+            <NavLink to="/help">Help</NavLink>
+          </li>
+          {!wasWatchingLocation ? (
+            <li>
+              <ShareMyLocationButton />
```

Comment:
Could do the same thing here: no ternary.
```suggestion
              <ShareMyLocationButton disabled={!wasWatchingLocation}/>
```

Reply:
Thanks! good point. I would like to say, we changed your suggestion about `!wasWatchingLocation` a little because the first time can be is disabled. we used only  `wasWatchingLocation` 😄 .